lib/async.zsh

```diff
@@ -23,12 +23,17 @@ typeset -g GEOMETRY_ASYNC_PROC_ID=0
     GEOMETRY_ASYNC_PROC_ID=$!
 }
 
-# Hooks precmd to run `-geometry_async_function` and hooks
-# `USR1` signal to comunicate with it
+# Removes the async temp file when zsh exits
+-geometry_async_zshexit() {
+    rm -f "${GEOMETRY_ASYNC_TMP_FULL_PATH}$$"
```

Comment:
`\` is done

I did remove the redirection, but that wasn't intentional. I can add it back if you believe we shouldn't report this error to the user.